kafka/consumer.go

```diff
@@ -261,6 +263,58 @@ func (c *Consumer) Events() chan Event {
 	return c.events
 }
 
+// ReadMessage polls the consumer for a message.
+//
+// This is a conveniance API that wraps Poll() and only returns
+// messages or errors. All other event types are discarded.
+//
+// The call will block for at most `timeoutMs` waiting for
+// a new message or error. `timeoutMs` may be set to -1 for
+// indefinate wait.
+//
+// Timeout is returned as (nil, err) where err is `kafka.(Error).Code == Kafka.ErrTimedOut`.
+//
+// Messages are returned as (msg, nil),
+// while general errors are returned as (nil, err),
+// and partition-specific errors are returned as (msg, err) where
+// msg.TopicPartition provides partition-specific information (such as topic, partition and offset).
+//
+// All other event types, such as PartitionEOF, AssignedPartitions, etc, are silently discarded.
+//
+func (c *Consumer) ReadMessage(timeoutMs int) (*Message, error) {
```

Comment:
any reason not to take a `timeout time.Duration` instead of an int?

Reply:
I can't seem to find a way to specify an infinite timeout with a Duration, and it is not nillable.
Ideas?

Reply:
should anyone really be using an infinite timeout in practice? (i.e. is this call cancellable by some mechanism?)

Reply:
The way it works now is 0 is infinite right? Could still work the same way. For example:

```
func main() {
	fn(0)
}

func fn(timeout time.Duration) {
	fmt.Printf("timeout: %v, %v", timeout, timeout == 0)
}

# >> timeout: 0s, true
```

Reply:
0 is no timeout, -1 is indefinate, > 0 is a standard timeout.
Same semantics as poll(2)

Reply:
If people wanted the timeout they'd generally call it by doing something like time.Second * 5

Reply:
Okay, so we should keep the current semantics with -1, 0, > 0, but use time.Duration instead?

Reply:
Yeah